packages/flutter/lib/src/scheduler/ticker.dart

```diff
@@ -12,80 +15,233 @@ import 'binding.dart';
 /// at the time of the callback being called.
 typedef void TickerCallback(Duration elapsed);
 
+/// An interface implemented by classes that can vend [Ticker] objects.
+abstract class TickerProvider {
+  /// Abstract const constructor. This constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
+  const TickerProvider();
+
+  /// Creates a ticker with the given callback.
+  ///
+  /// The kind of ticker provided depends on the kind of ticker provider.
```

Comment:
I think the text I have now is more grammatically correct. "implementing class" is a bit of a weird construction.

But more to the point, the text here isn't talking about the object type per se, but about the kind of behavior it may have. So for example, you could have a ticker provider return the same type as another, but with different behaviour. Case in point, most of the ticker providers essentially provide Ticker objects, but they don't all act the same.